src/main/scala/io/citrine/lolo/learners/ExtraRandomTrees.scala

```diff
@@ -0,0 +1,123 @@
+                             randomlyRotateFeatures: Boolean = false,
+                             rng: Random = Random
+                           ) extends Learner {
+    /**
+    * Train an extremely randomized tree ensemble model
+    *
+    * @param trainingData to train on
+    * @param weights      for the training rows, if applicable
+    * @return training result containing a model
+    */
+  override def train(trainingData: Seq[(Vector[Any], Any)], weights: Option[Seq[Double]]): TrainingResult = {
+    val breezeRandBasis: RandBasis = new RandBasis(new MersenneTwister(rng.nextLong()))
+
+    trainingData.head._2 match {
+      case _: Double =>
+        trainingData.head._2 match {
+          case _: Double =>
+            val numFeatures: Int = subsetStrategy match {
+              case x: String =>
+                x match {
+                  case "auto" => trainingData.head._1.size
+                  case "sqrt" => Math.ceil(Math.sqrt(trainingData.head._1.size)).toInt
+                  case "log2" => Math.ceil(Math.log(trainingData.head._1.size) / Math.log(2)).toInt
+                  case x: String =>
+                    println(s"Unrecognized subsetStrategy ${x}; using auto")
+                    trainingData.head._1.size
+                }
+              case x: Int =>
+                x
+              case x: Double =>
+                (trainingData.head._1.size * x).toInt
+            }
+            val DTLearner = RegressionTreeLearner(
+              leafLearner = leafLearner,
+              numFeatures = numFeatures,
+              minLeafInstances = minLeafInstances,
+              maxDepth = maxDepth,
+              splitter = ExtraRandomSplitter(rng)
+            )
+            val bagger = Bagger(if (randomlyRotateFeatures) FeatureRotator(DTLearner) else DTLearner,
+              numBags = numTrees,
+              useJackknife = useJackknife,
+              biasLearner = biasLearner,
+              uncertaintyCalibration = uncertaintyCalibration,
+              randBasis = breezeRandBasis
+            )
+            bagger.train(trainingData, weights)
+        }
+      case _: Any =>
+        trainingData.head._2 match {
+          case _: Double =>
+            val numFeatures: Int = subsetStrategy match {
+              case x: String =>
+                x match {
+                  case "auto" => Math.ceil(Math.sqrt(trainingData.head._1.size)).toInt
+                  case "sqrt" => Math.ceil(Math.sqrt(trainingData.head._1.size)).toInt
+                  case "log2" => Math.ceil(Math.log(trainingData.head._1.size) / Math.log(2)).toInt
+                  case x: String =>
+                    println(s"Unrecognized subsetStrategy ${x}; using auto")
+                    trainingData.head._1.size
+                }
+              case x: Int =>
+                x
+              case x: Double =>
+                (trainingData.head._1.size * x).toInt
+            }
+            val DTLearner = RegressionTreeLearner(
```

Comment:
This should be a classification tree learner.  The fact that its passing tests suggests that this code path is untested.